rust/noosphere/src/ffi.rs

```diff
@@ -0,0 +1,59 @@
+use std::collections::BTreeMap;
+
+use lazy_static::lazy_static;
+use safer_ffi::prelude::*;
+
+use crate::{
+    platform::{PlatformKeyMaterial, PlatformStore},
+    sphere::SphereContext,
+};
+
+lazy_static! {
+    static ref ACTIVE_SPHERES: BTreeMap<SphereHandle, Box<SphereContext<'static, PlatformKeyMaterial, PlatformStore>>> =
+        BTreeMap::new();
+}
+
+static mut GLOBAL_STORAGE_PATH: Option<String> = None;
+static mut SPHERE_STORAGE_PATH: Option<String> = None;
+
+ReprC! {
+    #[repr(C)]
+    pub struct SphereHandle {
+        user_identity: char_p::Box,
+        sphere_identity: char_p::Box,
+    }
+
+}
+
+ReprC! {
+    #[repr(C)]
+    pub struct SphereCreationResult {
+        mnemonic: char_p::Box
+    }
+}
+
+/// Set the global Noosphere storage path. This will be used to store user
+/// data and configuration that is cross-cutting with respect to sphere data
+#[ffi_export]
+pub fn noosphere_set_global_storage_path(path: char_p::Ref<'_>) {
+    unsafe {
+        GLOBAL_STORAGE_PATH = Some(path.to_string());
```

Comment:
Not yet relevant for the scope of the FFI interface, but are rust's mutable statics thread safe?

Reply:
Not on their own, you need to use a `Mutex` or something with effectively the same guarantees.